diff --git a/ceilometer/objectstore/swift.py b/ceilometer/objectstore/swift.py
index fbee19e..11f0635 100644
--- a/ceilometer/objectstore/swift.py
+++ b/ceilometer/objectstore/swift.py
@@ -31,6 +31,8 @@ from ceilometer.openstack.common import log
 from ceilometer.openstack.common import timeutils
 from ceilometer import plugin
 
+from urlparse import urljoin
+
 
 LOG = log.getLogger(__name__)
 
@@ -110,7 +112,8 @@ class SwiftPollster(_Base):
             LOG.debug(_("Swift endpoint not found"))
             return
 
-        base_url = '%s/v1/%s' % (endpoint, cfg.CONF.reseller_prefix)
+        base_url = urljoin(endpoint, '/v1/%s' % cfg.CONF.reseller_prefix)
+
         for t in ksclient.tenants.list():
             yield (t.id, swift.head_account('%s%s' % (base_url, t.id),
                                             ksclient.auth_token))
